numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -1473,6 +1489,12 @@ static PyObject *npyiter_itviews_get(NewNpyArrayIterObject *self)
         return NULL;
     }
 
+    if (self->managed == CONTEXT_EXITED) {
+        PyErr_SetString(PyExc_ValueError,
+                "Iterator is closed");
```

Comment:
Shouldn't this happen after `.close()` is called, not just after `__exit__` is called?

Reply:
yes, see new PR for issue #10950